docs/server-event-types.md

```diff
@@ -0,0 +1,49 @@
+---
+id: server-event-types
+title: Event type reference
+sidebar_label: Event types
+---
+
+Event types are defined in the Server API [event_type.proto](https://github.com/temporalio/api/blob/master/temporal/api/enums/v1/event_type.proto) file.
+
+- **EVENT_TYPE_UNSPECIFIED**:
+- **EVENT_TYPE_WORKFLOW_EXECUTION_STARTED**:
+- **EVENT_TYPE_WORKFLOW_EXECUTION_COMPLETED**:
+- **EVENT_TYPE_WORKFLOW_EXECUTION_FAILED**:
+- **EVENT_TYPE_WORKFLOW_EXECUTION_TIMED_OUT**:
```

Comment:
- **EVENT_TYPE_UNSPECIFIED**:
This event type is a place holder, will never appear in user's workflow execution history

- **EVENT_TYPE_WORKFLOW_EXECUTION_STARTED**:
This event type indicates the workflow execution is started / triggered.
This event type contains workflow execution inputs, as well as workflow timeout configurations.

- **EVENT_TYPE_WORKFLOW_EXECUTION_COMPLETED**:
This event type indicates the workflow execution is successfully completed.
This event type contains workflow execution results.

- **EVENT_TYPE_WORKFLOW_EXECUTION_FAILED**:
This event type indicates the workflow execution is unsuccessfully completed.
This event type contains workflow execution error.

- **EVENT_TYPE_WORKFLOW_EXECUTION_TIMED_OUT**:
This event type indicates the workflow execution is timed out by Temporal server, due to workflow not completed within timeout settings.

- **EVENT_TYPE_WORKFLOW_EXECUTION_TERMINATED**:
This event type indicates the workflow execution is forcefully terminated, e.g. caller call terminate workflow API on the Temporal server side. 

**EVENT_TYPE_WORKFLOW_EXECUTION_CANCELED**:
This event type indicates the workflow execution is cancelled, e.g. caller call cancel workflow API on the Temporal server side, Temporal server let SDK know workflow should be cancelled, SDK decided to cancel workflow.

- **EVENT_TYPE_WORKFLOW_EXECUTION_CONTINUED_AS_NEW**:
This event type indicates workflow is successfully completed, and a new workflow should be started within the same transaction.
This event type contains last workflow execution results, as well as new workflow execution inputs.